packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -416,6 +416,10 @@ abstract class VpcBase extends Resource implements IVpc {
     let subnets = allSubnets[subnetType];
 
     if (onePerAz && subnets.length > 0) {
+      subnets = subnets.filter(s => s.availabilityZone);
```

Comment:
I had rather you not do this, but make `availabilityZone` throw an exception if it's read but not set.